File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example_py/h10-w/chassis/get_odometry_message.py
import humanoid_sdk_py
import humanoid_sdk_py.h10w as h10w

print("=== Get Chassis Odometry Message ===")
h10w_chassis = h10w.H10wChassis()
ret, tuple_data = h10w_chassis.getOdometryMessage()
if ret != 0:
    print("get odometry message failed.")
    exit(-1)

cur_time, cur_mileage = tuple_data
print("get odometry message success.")
print(
    f"chassis odometry message: current time:{cur_time}, current mileage:{cur_mileage}"
)

File: gr00t/Isaac-GR00T/examples/H20/robots/sdk_package/example/h10-w/chassis/get_odometry_message.cpp
// get_odometry_message.cpp

#include <iostream>
#include <iomanip>
#include "h10-w/h10w_chassis.h"
#include "common/common_def.h"

using namespace humanoid;
using namespace humanoid::h10w;

int main() {
  std::cout << "=== 获取里程计信息示例 ===" << std::endl;
  H10wChassis chassis;
  int cur_time;
  int cur_mileage;

  if (chassis.getOdometryMessage(cur_time, cur_mileage) == 0) {
    std::cout << "获取成功" << std::endl;
    std::cout << "    累计运行时间=" << cur_time << "s, 累计里程="
              << cur_mileage << "m" << std::endl;
  } else {
    std::cout << "获取失败" << std::endl;
    return -1;
  }
  return 0;
}
